storage/cloud-client/encryption.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to upload and download encrypted blobs
+(objects) in Google Cloud Storage.
+
+Use `generate-encryption-key` to generate an example key:
+
+    python encryption.py generate-encryption-key
+
+Then use the key to upload and download files encrypted with a custom key.
+
+For more information, see the README.md under /storage and the documentation
+at https://cloud.google.com/storage/docs/encryption.
+"""
+
+import argparse
+import base64
+import os
+
+from gcloud import storage
+
+
+def generate_encryption_key():
+    """Generates a 256 bit (32 byte) AES encryption key and prints the
+    base64 representation.
+
+    This is included for demonstration purposes. You should generate your own
+    key. Please remember that encryption keys should be handled with a
+    comprehensive security policy.
+    """
+    key = os.urandom(32)
+    encoded_key = base64.b64encode(key).decode('utf-8')
+    print('Base 64 encoded encryption key: {}'.format(encoded_key))
+
+
+def upload_encrypted_blob(bucket_name, source_file_name,
+                          destination_blob_name, base64_encryption_key):
+    """Uploads a file to a Google Cloud Storage bucket using a custom
+    encryption key.
+
+    The file will be encrypted by Google Cloud Storage and only
+    retrievable using the provided encryption key.
+    """
+    storage_client = storage.Client()
+    bucket = storage_client.get_bucket(bucket_name)
+    blob = bucket.blob(destination_blob_name)
+
+    # Encryption key must be an AES256 key represented as a bytestring with
+    # 32 bytes. Since it's passed in as a base64 encoded string, it needs
```

Comment:
Per our chat - I'm debating if the fact that you use base64 could throw people off. It's a side-effect of the way you produced the key, and not a requirement of the API.

Maybe update this comment as follows? "Since this sample application uses a base64 encoded string to store our encryption key, it needs..."

That way you can make it clear it's just a design choice, and not a functional requirement.